tests/com/rcv/TabulatorTests.java

```diff
@@ -89,33 +90,36 @@ static boolean fileCompare(String path1, String path2) {
     return result;
   }
 
+  // function: getConfigPath
+  // purpose: given stem returns path to config file in test asset folder
+  // returns: path to config file
+  static String getConfigPath(String stem) {
+    return Paths.get(System.getProperty("user.dir"),
+      TEST_ASSET_FOLDER, stem,
+      stem + "_config.json")
+      .toAbsolutePath()
+      .toString();
+  }
+
   // function: runTabulationTest
   // purpose: helper function to support running various tabulation tests
   // param: stem base name of folder containing config file cvr files and expected result files
   static void runTabulationTest(String stem) {
-    // full path to config file
-    String configPath = Paths.get(System.getProperty("user.dir"),
-        TEST_ASSET_FOLDER, stem,
-        stem + "_config.json")
-        .toAbsolutePath()
-        .toString();
     // full path to expected results file
     String expectedPath = Paths.get(System.getProperty("user.dir"),
         TEST_ASSET_FOLDER,
         stem,
         stem + "_expected.json")
         .toAbsolutePath()
         .toString();
-    // we use config file parent folder as default for resolving user paths
-    FileUtils.setUserDirectory(new File(configPath).getParent());
-    // load the contest config
-    ContestConfig config = Main.loadContestConfig(configPath);
-    Assertions.assertNotNull(config);
-    executeTabulation(config);
+
+    // create a session object and run the tabulation
+    TabulatorSession session = new TabulatorSession(TabulatorTests.getConfigPath(stem));
+    session.tabulate();
     // actualSummaryOutputPath is the summary json we just tabulated
-    String actualSummaryOutputPath = Main.getSummaryOutputPath();
+    String actualSummaryOutputPath = session.summaryOutputPath;
     // compare actual to expected
-    assertTrue(fileCompare(actualSummaryOutputPath, expectedPath));
+    assertTrue(fileCompare(expectedPath, actualSummaryOutputPath));
```

Comment:
Since we're actually creating files here, I'd strongly recommend adding logic here that cleans up after the test is complete, e.g. a `teardown` function that removes all created files and directories after each test is run.

It's also really hard to parse out the results of each individual test given all the log / console window spam; I'd think the testing framework should allow us to only show the test results in a single output / console window? I guess it's not really a problem as long as we're running things inside of IntelliJ, but maybe something to add to the wishlist.

I'm not sure about the whole `junit.jupiter` sub-library; why are we using that instead of the basic `junit` suite?

Reply:
Yeah, I was thinking about something like that too... the scale test does produce a lot of data.  I think automated cleanup would be fine and just skip cleanup when the test fails so we can look at it.  I will add that.  

As for output, yes there is a visualizer for this.  Did you follow the directions here?  https://github.com/BrightSpots/rcv/pull/194

It is using junit5, and you should see something like this when running tests from IntelliJ:

<img width="1440" alt="regressiontests" src="https://user-images.githubusercontent.com/838732/51554280-c5a61880-1e29-11e9-9620-9cd38c028df6.png">

Reply:
Clean up sounds great!

Yep, sorry I wasn't more clear: I do have that working in IntelliJ, but I was thinking about someone running the tests outside of IntelliJ, like from the CLI. Just my reluctance to rely on third party software to develop effectively. Anyway, this isn't necessary to worry about now since the three of us are all using IntelliJ, but I think it's worth filing an issue to make it more friendly to other devs who might not be in the future.

Reply:
Gotcha.  It really makes sense to use a thrid party tool here.  I researched this when I began planning the test code and there are various mature options.  I picked Junit5 because it is modern, widely used, well supported, and offers a rich feature set, although I expect we will only use the basics.  It can be integrated with other IDEs, and run standalone as well.  IMO it is pretty developer friendly.  https://stackoverflow.com/questions/52373469/how-to-launch-junit-5-platform-from-the-command-line-without-maven-gradle/52373592#52373592